### PR TITLE
fixes critical vulnerabilities on ray

### PR DESCRIPTION
**Pull request by spolti**
chore: 	fix the following CVEs
	- [CVE-2023-6019](https://www.cve.org/CVERecord?id=CVE-2023-6019): Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')
	- [CVE-2023-6020](https://www.cve.org/CVERecord?id=CVE-2023-6020): Use of GET Request Method With Sensitive Query Strings

There are no fix for [CVE-2023-48023](https://www.cve.org/CVERecord?id=CVE-2023-48023) yet

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by spolti** [comment]
The failing test reports only:

```
ValueError: `detached=False` is no longer supported. In a future release, it will be removed altogether.
```

on .py files like

**Comment by spolti** [comment]
The failing image build has succeed locally.

on GHA it is failing due:   `FileNotFoundError: [Errno 2] No such file or directory: '/prod_venv/lib/python3.9/site-packages/idna-3.6.dist-info'`

not sure what might be missing.

**Comment by yuzisun** [comment]
@spolti Can you help rebase?

**Comment by spolti** [comment]
> @spolti Can you help rebase?

Hi @yuzisun, updated, sry for the delay.

**Review by israel-hdez — APPROVED** [review]
I no longer see a failed run. So, I leave my approval.

**Comment by terrytangyuan** [comment]
@spolti nitpick: can you fix the typo in the PR title?

**Review by terrytangyuan — COMMENTED** [review]
LGTM

File: python/kserve/pyproject.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ kubernetes = ">=23.3.0"
 python-dateutil = "^2.8.0"
 numpy = "^1.23.5"
 psutil = "^5.9.0"
-ray = { version = "^2.8.0", extras = ["serve"] }
+ray = { version = ">=2.8.1", extras = ["serve"] }
```
Comment (ddelange) [inline python/kserve/pyproject.toml:45]: hi :wave: why was the upper cap removed?

poetry `^2.8.1` == pip `>=2.8.1,<3`

Comment (ddelange) [reply python/kserve/pyproject.toml:45]: also, to propagate a version bump in the base package to the dependent packages' lockfiles, `make poetry-update-lockfiles` was recently added (and would be needed here)

Comment (sivanantha321) [reply python/kserve/pyproject.toml:45]: > hi 👋 why was the upper cap removed?
> 
> poetry `^2.8.1` == pip `>=2.8.1,<3`

Fixed here #3389

Comment (ddelange) [reply python/kserve/pyproject.toml:45]: awesome!

Comment (spolti) [reply python/kserve/pyproject.toml:45]: many thanks @sivanantha321

**Pull request merged**